pkg/diagnostics/cluster/aggregated_logging/services.go

```diff
@@ -0,0 +1,49 @@
+package aggregated_logging
+
+import (
+	"fmt"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+var loggingServices = sets.NewString("logging-es", "logging-es-cluster", "logging-es-ops", "logging-es-ops-cluster", "logging-kibana", "logging-kibana-ops")
+
+const serviceNotFound = `
+Expected to find '%s' among the logging services for the project but did not.  Depending 
+upon the options chosen during installation, these services (e.g. logging-es-ops) may not have been provided.
+`
+
+// checkServices looks to see if the aggregated logging services exist
+func checkServices(r diagnosticReporter, adapter servicesAdapter, project string) {
+	r.Info("AGL0200", fmt.Sprintf("Checking for services in project '%s' with selector '%s'", project, loggingSelector.AsSelector()))
+	serviceList, err := adapter.services(project, kapi.ListOptions{LabelSelector: loggingSelector.AsSelector()})
+	if err != nil {
+		r.Error("AGL0205", err, fmt.Sprintf("There was an error while trying to retrieve the logging services: %s", err))
+		return
+	}
+	foundServices := sets.NewString()
+	for _, service := range serviceList.Items {
+		foundServices.Insert(service.ObjectMeta.Name)
+		r.Debug("AGL0210", fmt.Sprintf("Retrieved service '%s'", service.ObjectMeta.Name))
+	}
+	for _, service := range loggingServices.List() {
+		if foundServices.Has(service) {
+			checkServiceEndpoints(r, adapter, project, service)
+		} else {
+			r.Warn("AGL0215", nil, fmt.Sprintf(serviceNotFound, service))
```

Comment:
I don't like that all the services are treated as equal here. The non-ops ones should be an error if they're missing, as nothing will work. The ops ones should be errors too, if we can determine that they're supposed to exist, but I'm not sure we can cleanly, and I think the deployment creates them regardless of whether there's anything to use them. So I think the ops ones should just be warnings if they're missing.